src/client/Microsoft.Identity.Client/Extensibility/AcquireTokenForClientBuilderExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Identity.Client.Extensibility
+{
+
+    /// <summary>
+    /// Extensions for <see cref="AcquireTokenForClientBuilderExtensions"/> class
+    /// </summary>
+    public static class AcquireTokenForClientBuilderExtensions
+    {
+        /// <summary>
+        /// Binds the token to a key in the cache. L2 cache keys contain the key id.
+        /// No cryptographic operations is performed on the token.
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <param name="keyId">A key id to which the access token is associated. The token will not be retrieved from the cache unless the same key id is presented. Can be null.</param>
+        /// <returns></returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        public static AcquireTokenForClientParameterBuilder WithKeyId(
```

Comment:
@jmprieur - second part of extensibility

@pmaytak @trwalke - this was already reviewed as part of the other PR, but I did not find a good way to commit just this part. Apollogies for the churn.

Reply:
This is a public API?

Reply:
Yes, this is a public api, hidden in the "extensibility" namespace. I want to avoid "friend assemblies" because internal APIs are easily broken.